src/Glpi/Form/Migration/FormMigration.php

```diff
@@ -454,4 +517,86 @@ private function updateBlockHorizontalRank(): void
             );
         }
     }
+
+    private function processMigrationOfAccessControls(): void
+    {
+        $this->progress_indicator?->setProgressBarMessage(__('Importing access controls...'));
+
+        // Retrieve data from glpi_plugin_formcreator_forms table
+        $raw_form_access_rights = $this->db->request([
+            'SELECT' => [
+                'access_rights',
+                new QueryExpression('glpi_plugin_formcreator_forms.id', 'forms_id'),
+                'name', // Added to get form name for status reporting
+                new QueryExpression('JSON_ARRAYAGG(users_id)', 'user_ids'),
+                new QueryExpression('JSON_ARRAYAGG(groups_id)', 'group_ids'),
+                new QueryExpression('JSON_ARRAYAGG(profiles_id)', 'profile_ids')
+            ],
+            'FROM'   => 'glpi_plugin_formcreator_forms',
+            'LEFT JOIN'   => [
+                'glpi_plugin_formcreator_forms_users' => [
+                    'ON' => [
+                        'glpi_plugin_formcreator_forms_users' => 'plugin_formcreator_forms_id',
+                        'glpi_plugin_formcreator_forms'       => 'id'
+                    ]
+                ],
+                'glpi_plugin_formcreator_forms_groups' => [
+                    'ON' => [
+                        'glpi_plugin_formcreator_forms_groups' => 'plugin_formcreator_forms_id',
+                        'glpi_plugin_formcreator_forms'        => 'id'
+                    ]
+                ],
+                'glpi_plugin_formcreator_forms_profiles' => [
+                    'ON' => [
+                        'glpi_plugin_formcreator_forms_profiles' => 'plugin_formcreator_forms_id',
+                        'glpi_plugin_formcreator_forms'          => 'id'
+                    ]
+                ]
+            ],
+            'GROUPBY' => ['forms_id', 'access_rights']
+        ]);
+
+        foreach ($raw_form_access_rights as $form_access_rights) {
+            $form = Form::getById(
+                $this->getMappedItemTarget(
+                    'PluginFormcreatorForm',
+                    $form_access_rights['forms_id']
+                )['items_id'] ?? 0
+            );
+            if ($form === false) {
+                throw new LogicException("Form with id {$form_access_rights['forms_id']} not found");
+            }
+
+            $strategy_class = self::getStrategyForAccessTypes()[$form_access_rights['access_rights']] ?? null;
+            if ($strategy_class === null) {
+                throw new LogicException("Strategy class not found for access type {$form_access_rights['access_rights']}");
+            }
+
+            $manager = FormAccessControlManager::getInstance();
```

Comment:
It would be preferable to inject it in your migration constructor. This would require changing the `getMigrationClass(): string` method by a `getMigration(): AbstractPluginMigration`  method.